crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1770,11 +1780,30 @@ impl<'db> TypeInferenceBuilder<'db> {
 
             (Type::StringLiteral(lhs), Type::StringLiteral(rhs), ast::Operator::Add) => {
                 Type::StringLiteral(StringLiteralType::new(self.db, {
-                    let lhs_value = lhs.value(self.db);
-                    let rhs_value = rhs.value(self.db);
-                    lhs_value.clone() + rhs_value
+                    let lhs_value = lhs.value(self.db).to_string();
+                    let rhs_value = rhs.value(self.db).as_ref();
+                    (lhs_value + rhs_value).into()
```

Comment:
Can't this result in string literals that are larger than `MAX_STRING_LITERAL_SIZE`?